### PR TITLE
feat(metrics): allow editing of effect sizes

### PR DESCRIPTION
**Pull request by anotherchrisberry**
They've been visible in the UI but not editable for some time. We have enough people using them that it makes sense to make them editable now. 

Because people think of the effect sizes as percentages, we're handling the conversion in the UI. In the future, if we are going to add a feature like effect sizes, I would encourage the API designers to consider how people will think of these values and build the API accordingly, but that ship has long sailed on this one.

There's also a new measurement type (CLES) available in Kayenta, but we are not going to allow users to select it via the UI (per @csanden). However, if they've already added it into their config, we'll let them toggle it. It adds quite a bit of complexity to an otherwise not very complex component.

There are a handful of other small changes that are just there to properly align form fields and labels. It's been driving me a little crazy how they are misaligned.

Available inputs get toggled based on the direction of the metric: when it's "either", we show everything, when it's "increase" or "decrease", we only show those respective fields.

**Failing on increase, only showing increase fields**
Screen Shot 2020-08-17 at 10 46 57 PM

**Failing on decrease, only showing decrease fields**
Screen Shot 2020-08-17 at 10 46 50 PM

**Fail either way**
Screen Shot 2020-08-17 at 10 46 42 PM

**You can use CLES!** But you shouldn't, we have no idea how it works
Screen Shot 2020-08-17 at 10 53 35 PM

### CODE REVIEW

File: settings.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ window.spinnakerSettings = {
     atlasWebComponentsUrl: atlasWebComponentsUrl,
     atlasWebComponentsPolyfillUrl: atlasWebComponentsPolyfillUrl,
     templatesEnabled: templatesEnabled,
-    showAllConfigs: true,
+    showAllConfigs: false,
```
Comment (anotherchrisberry) [inline settings.js:104]: I have no idea why this was the local config value but let's not do it?

File: src/kayenta/actions/creators.ts [hunk 2]
```diff
@@ -38,6 +39,9 @@ export const updateMetricNanStrategy = createAction<{ id: string; strategy: stri
 export const updateMetricCriticality = createAction<{ id: string; critical: boolean }>(
   Actions.UPDATE_METRIC_CRITICALITY,
 );
+export const updateEffectSize = createAction<{ id: string; value: ICanaryMetricEffectSizeConfig }>(
+  Actions.UPDATE_EFFECT_SIZE,
+);
```
Comment (anotherchrisberry) [inline src/kayenta/actions/creators.ts:44]: I'm finally starting to understand redux a little bit and I don't hate it as much as I did when I didn't understand it at all.

File: src/kayenta/layout/formRow.tsx [hunk 3]
```diff
@@ -9,18 +9,23 @@ export interface IFormRowProps {
   checkbox?: boolean;
   error?: string;
   warning?: string;
+  inputOnly?: boolean;
 }
 
-export default function FormRow({ label, helpId, children, checkbox, error, warning }: IFormRowProps) {
+export default function FormRow({ label, helpId, children, checkbox, error, warning, inputOnly }: IFormRowProps) {
```
Comment (anotherchrisberry) [inline src/kayenta/layout/formRow.tsx:15]: This is to fix all the misaligned form field labels

File: src/kayenta/reducers/selectedConfig.ts [hunk 4]
```diff
@@ -111,9 +111,9 @@ const metricList = handleActions(
     [Actions.ADD_METRIC]: (state: ICanaryMetricConfig[], action: Action & any) =>
       idMetrics(state.concat([action.payload.metric])),
     [Actions.REMOVE_METRIC]: (state: ICanaryMetricConfig[], action: Action & any) =>
-      idMetrics(state.filter(metric => metric.id !== action.payload.id)),
+      idMetrics(state.filter((metric) => metric.id !== action.payload.id)),
```
Comment (anotherchrisberry) [inline src/kayenta/reducers/selectedConfig.ts:114]: Did prettier change its default wrapping of single values? Seems like it! That's fine! I'll submit a followup PR to re-prettier everything.

File: index.deck [hunk 5]
```diff
@@ -1,99 +1,152 @@
 <!doctype html>
-<html class="no-js" ng-app="netflix.spinnaker">
+<html class="no-js" ng-app="netflix.spinnaker" ng-strict-di="true">
```
Comment (anotherchrisberry) [inline index.deck:2]: clearly unrelated, I just copied the file out of the current deck repo

File: src/kayenta/reducers/selectedConfig.ts (hunk 4 above)
Comment (maggieneterval) [reply src/kayenta/reducers/selectedConfig.ts:114]: [I think so!](https://prettier.io/docs/en/options.html#arrow-function-parentheses)

**Pull request merged**